### PR TITLE
Fix warnings in Query class

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java [hunk 1]
```diff
@@ -545,14 +545,14 @@ private synchronized void closeExchangeClientIfNecessary(QueryInfo queryInfo)
...
-    private void handleSerializationException(Throwable exception)
+    private synchronized void handleSerializationException(Throwable exception)
     {
         // failQuery can throw exception if query has already finished.
         try {
             queryManager.failQuery(queryId, exception);
         }
         catch (RuntimeException e) {
-            log.debug("Could not fail query", e);
+            log.debug(e, "Could not fail query");
```
Comment (zinking) [inline core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: why this change? this doesn't look right,

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java [hunk 2]
```diff
@@ -633,23 +633,23 @@ private String formatType(DataType type)
                 if (dataTimeType.getType() == DateTimeDataType.Type.TIMESTAMP && !dataTimeType.isWithTimeZone()) {
                     return TIMESTAMP;
                 }
-                else if (dataTimeType.getType() == DateTimeDataType.Type.TIME && !dataTimeType.isWithTimeZone()) {
+                if (dataTimeType.getType() == DateTimeDataType.Type.TIME && !dataTimeType.isWithTimeZone()) {
```
Comment (zinking) [inline core/trino-main/src/main/java/io/trino/server/protocol/Query.java:636]: ?

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java [hunk 3]
```diff
@@ -545,14 +545,14 @@ private synchronized void closeExchangeClientIfNecessary(QueryInfo queryInfo)
...
 
-    private void handleSerializationException(Throwable exception)
+    private synchronized void handleSerializationException(Throwable exception)
     {
         // failQuery can throw exception if query has already finished.
         try {
             queryManager.failQuery(queryId, exception);
         }
         catch (RuntimeException e) {
-            log.debug("Could not fail query", e);
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: we should have static analysis to catch that.

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java (hunk 1 above)
Comment (electrum) [reply core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: Good observation. This is actually a bug fix, found by IntelliJ warnings. Our logger API takes the exception as the first argument: https://github.com/airlift/airlift/blob/5849abe7466b007ec94d4c388918983172703fbd/log/src/main/java/io/airlift/log/Logger.java#L65

Before this change, the exception is an argument to the formatted message, which is wrong since the message doesn't have a placeholder for it (and it wouldn't get logged properly as an exception): https://github.com/airlift/airlift/blob/5849abe7466b007ec94d4c388918983172703fbd/log/src/main/java/io/airlift/log/Logger.java#L93

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java (hunk 3 above)
Comment (electrum) [reply core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: IntelliJ found this one. I created https://github.com/airlift/airlift/pull/922 for use with https://errorprone.info/bugpattern/FormatStringAnnotation

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java (hunk 2 above)
Comment (electrum) [reply core/trino-main/src/main/java/io/trino/server/protocol/Query.java:636]: The "else" is redundant since the above "if" always returns. IntelliJ will warn and perform this conversion automatically (and safely).

File: core/trino-main/src/main/java/io/trino/server/protocol/Query.java (hunk 1 above)
Comment (zinking) [reply core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: I guess people (at least me + the original line author) are just used to `log.info("xxxx {}", param1, exception)` logging paradigm, so I would suggest fix from where this is rooted (airlift).

Comment (electrum) [reply core/trino-main/src/main/java/io/trino/server/protocol/Query.java:555]: @zinking the Airlift logger API was intentionally designed to have the exception first, since that allows using it unambiguously with string formatting arguments. We can catch it with [Error Prone](https://errorprone.info/bugpattern/FormatStringAnnotation) after merging https://github.com/airlift/airlift/pull/922
